src/core/authenticator.cpp

```diff
@@ -27,6 +27,12 @@ using namespace std;
 Authenticator::Authenticator(const Config &config) {
     mysql_init(&con);
     Log::log_with_date_time("connecting to MySQL server " + config.mysql.server_addr + ':' + to_string(config.mysql.server_port), Log::INFO);
+    if (config.mysql.cafile != "") {
+        mysql_ssl_set(&con, NULL, NULL, config.mysql.cafile.c_str(), NULL, NULL);
+    }
+    if (config.mysql.tls_version == "") {
```

Comment:
Do you really mean `tls_version == ""`?

Reply:
https://mariadb.com/kb/en/configuring-mariadb-connectorc-with-option-files/#tls_version

Reply:
My fault. I have fixed it.